cpp/include/cuml/manifold/common.hpp

```diff
@@ -110,10 +111,11 @@ struct manifold_precomputed_knn_inputs_t : public manifold_inputs_t<value_t> {
   {
     // Return true if data is on CPU (need to allocate device memory)
     // Return false if data is already on device (no allocation needed)
-    auto pointer_residency = raft::spatial::knn::detail::utils::check_pointer_residency(
-      knn_graph.knn_indices, knn_graph.knn_dists);
-    return pointer_residency == raft::spatial::knn::detail::utils::pointer_residency::host_only ||
-           pointer_residency == raft::spatial::knn::detail::utils::pointer_residency::mixed;
+    auto indices_mem_type = raft::memory_type_from_pointer(knn_graph.knn_indices);
+    auto dists_mem_type   = raft::memory_type_from_pointer(knn_graph.knn_dists);
+
+    return !raft::is_device_accessible(indices_mem_type) ||
+           !raft::is_device_accessible(dists_mem_type);
```

Comment:
We are tracking an issue in CI that makes the CUDA context invalid. It looks like it may be an illegal memory accesses that happens when UMAP is given a pre-computed KNN on host memory while the HMM feature enabled (making host pointers device accessible). We disabled pre-computed KNN on host from now, but ideally we would want to enable it while disabling it in the specific case of HMM enabled (we need to investigate the HMM case separately).